examples/tls/src/tests.rs

```diff
@@ -1,8 +1,40 @@
-use rocket::local::blocking::Client;
+use rocket::{http::tls::Certificate, local::blocking::Client};
 
 #[test]
-fn hello_world() {
+fn get_index_without_auth() {
     let client = Client::tracked(super::rocket()).unwrap();
     let response = client.get("/").dispatch();
-    assert_eq!(response.into_string(), Some("Hello, world!".into()));
+    assert_eq!(response.into_string(), Some("Hello, anonymous user".into()));
+}
+
+#[test]
+fn get_secret_without_auth() {
+    let client = Client::tracked(super::rocket()).unwrap();
+    let response = client.get("/secret").dispatch();
+    assert_eq!(response.status().code, 404);
+}
+
+fn get_cert() -> Certificate {
+    let sans = ["mary".to_string()];
+    let cert = rcgen::generate_simple_self_signed(sans.to_vec()).unwrap();
+    let cert = cert.serialize_der().unwrap();
+    Certificate(cert)
+}
```

Comment:
We shouldn't ask users to figure out how to generate a simple one of these. We should provide methods on `Certificate` to do this much more easily.